File: app/src/main/java/com/charlie/sawl/widgets/timetable/TimeTableWidgetService.java
package com.charlie.sawl.widgets.timetable;

import android.content.Intent;
import android.widget.RemoteViewsService;

public class TimeTableWidgetService extends RemoteViewsService {

    @Override
    public RemoteViewsFactory onGetViewFactory(Intent intent) {
        return (new TimeTableFactory(this.getApplicationContext(), intent));
    }
}

File: app/src/main/kotlin/com/charlie/sawl/widgets/timetable/TimetableWidgetService.kt
package com.charlie.sawl.widgets.timetable

import android.content.Intent
import android.widget.RemoteViewsService

class TimetableWidgetService : RemoteViewsService() {

    override fun onGetViewFactory(intent: Intent): RemoteViewsService.RemoteViewsFactory {
        return TimetableFactory(this.applicationContext, intent)
    }
}